tests/e2e/tikv/tikv.go

```diff
@@ -0,0 +1,545 @@
+		scaleInParallelism: 1,
+		originDeleteSlots:  sets.NewInt32(1),
+		finalDeleteSlots:   sets.NewInt32(),
+		scaleInOrdinals:    []int32{5, 4, 3},
+		deleteSlotsEvolutions: [][]int32{
+			{1},
+		},
+		scaleInGroups: [][]int32{
+			{5},
+			{4},
+			{3},
+		},
+	},
+	{
+		idx:                3,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1),
+		finalDeleteSlots:   sets.NewInt32(),
+		scaleInOrdinals:    []int32{5, 4, 3},
+		deleteSlotsEvolutions: [][]int32{
+			{1},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{3},
+		},
+	},
+	{
+		idx:                4,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 1,
+		originDeleteSlots:  sets.NewInt32(1),
+		finalDeleteSlots:   sets.NewInt32(2),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1},
+			{2},
+		},
+		scaleInGroups: [][]int32{
+			{5},
+			{4},
+			{2},
+		},
+	},
+	{
+		idx:                5,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 3,
+		originDeleteSlots:  sets.NewInt32(1),
+		finalDeleteSlots:   sets.NewInt32(2),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1},
+			{2},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4, 2},
+		},
+	},
+	{
+		idx:                6,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1, 2),
+		finalDeleteSlots:   sets.NewInt32(2, 3),
+		scaleInOrdinals:    []int32{6, 5, 3, 4},
+		deleteSlotsEvolutions: [][]int32{
+			{1, 2},
+			{2},
+			{2, 3},
+		},
+		scaleInGroups: [][]int32{
+			{6, 5},
+			{3},
+		},
+	},
+	{
+		idx:                7,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1),
+		finalDeleteSlots:   sets.NewInt32(2, 6),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1},
+			{6},
+			{2, 6},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{2},
+		},
+	},
+	{
+		idx:                8,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1, 6),
+		finalDeleteSlots:   sets.NewInt32(2),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1, 6},
+			{2},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{2},
+		},
+	},
+	{
+		idx:                9,
+		originTikvReplica:  5,
+		finalTikvReplica:   3,
+		scaleInParallelism: 2,
+		originDeleteSlots:  sets.NewInt32(1, 6),
+		finalDeleteSlots:   sets.NewInt32(2, 7),
+		scaleInOrdinals:    []int32{5, 4, 2},
+		deleteSlotsEvolutions: [][]int32{
+			{1, 6},
+			{7},
+			{2, 7},
+		},
+		scaleInGroups: [][]int32{
+			{5, 4},
+			{2},
+		},
+	},
+}
+
+func (t *testcase) description() string {
+	return fmt.Sprintf("[origin-%v-final-%v-parallelism-%v-testcase-idx-%v]", t.originTikvReplica, t.finalTikvReplica, t.scaleInParallelism, t.idx)
+}
```

Comment:
I tried, but I found we can only put all the information like replicas and delete-slots of origin and final to distinguish. 
Would it be a good try ? 
Or maybe any other idea ?

Reply:
name is hard 🤣 , how about `description(index int)` and passing the index of list, at least we don't need to maintain index

Reply:
Yeah I also used the index of list at first time, I change it by manually since when I debugging some failed case by comment other cases out, and it would become quiet annoying to locate the test by manually count the index..

If you still think it's supposed to be removed I can do it, but I think keep it would be more convenient to locate a failed case.